src/CSnakes.Runtime/CPython/Init.cs

```diff
@@ -126,28 +170,48 @@ private void InitializeEmbeddedPython()
     [LibraryImport(PythonLibraryName, EntryPoint = "Py_SetPath", StringMarshallingCustomType = typeof(Utf32StringMarshaller), StringMarshalling = StringMarshalling.Custom)]
     internal static partial void Py_SetPath_UCS4_UTF32(string path);
 
+    protected void FinalizeEmbeddedPython()
+    {
+        lock (initLock)
+        {
+            if (!IsInitialized)
+                return;
+
+            // Shut down asyncio coroutines
+            CloseEventLoops();
+
+            // Clean-up interns
+            NewEventLoopFactory?.Dispose();
+            AsyncioModule?.Dispose();
+            // TODO: Add more cleanup code here

```

Comment:
This TODO is to clean up the currently static globals which we never dispose of, but are cleaned up automatically. We should dispose of them incase people recycle environments.